homeassistant/components/ovo_energy/sensor.py

```diff
@@ -0,0 +1,212 @@
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+    client: OVOEnergy = hass.data[DOMAIN][entry.entry_id][DATA_CLIENT]
+
+    now = datetime.utcnow()
+    try:
+        usage: OVODailyUsage = await client.get_daily_usage(now.strftime("%Y-%m"))
+        currency = usage.electricity[len(usage.electricity) - 1].cost.currency_unit
+    except aiohttp.ClientError as exception:
+        _LOGGER.warning(exception)
+        raise PlatformNotReady from exception
+
+    async_add_entities(
+        [
+            OVOEnergyLastElectricityReading(coordinator, client),
+            OVOEnergyLastGasReading(coordinator, client),
+            OVOEnergyLastElectricityCost(coordinator, client, currency),
+            OVOEnergyLastGasCost(coordinator, client, currency),
+        ],
+        True,
+    )
+
+
+class OVOEnergySensor(OVOEnergyDeviceEntity):
+    """Defines a OVO Energy sensor."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        client: OVOEnergy,
+        key: str,
+        name: str,
+        icon: str,
+        unit_of_measurement: str = "",
+    ) -> None:
+        """Initialize OVO Energy sensor."""
+        self._state = None
+        self._attributes = None
+        self._unit_of_measurement = unit_of_measurement
+
+        super().__init__(coordinator, client, key, name, icon)
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self) -> object:
+        """Return the attributes of the sensor."""
+        return self._attributes
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+
+class OVOEnergyLastElectricityReading(OVOEnergySensor):
+    """Defines a OVO Energy last reading sensor."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, client: OVOEnergy):
+        """Initialize OVO Energy sensor."""
+
+        super().__init__(
+            coordinator,
+            client,
+            f"{client.account_id}_last_electricity_reading",
+            "OVO Last Electricity Reading",
+            "mdi:flash",
+            "kWh",
+        )
+
+    async def _ovo_energy_update(self) -> bool:
+        """Update OVO Energy entity."""
+        usage: OVODailyUsage = self._coordinator.data
+        if usage is None or usage.electricity is None:
+            _LOGGER.warning("No data found for %s", self._name)
+            return False
+        last_reading: OVODailyElectricity = usage.electricity[
+            len(usage.electricity) - 1
+        ]
+        self._state = last_reading.consumption
+        self._attributes = {
+            "start_time": last_reading.interval.start,
+            "end_time": last_reading.interval.end,
+        }
+        return True
```

Comment:
Considering earlier suggestion for putting the coordinator in charge, that allows to do something like this:

```py
    @property
    def state(self) -> str:
        """Return the state of the sensor."""
        usage: OVODailyUsage = self._coordinator.data
        if usage is None or not usage.electricity:
            return None
        return usage.electricity[-1].consumption
```

This can be applied to other sensors in this PR as well.

Also changed `usage.electricity is None` -> `not usage.electricity` as that deals with the possibility when `electricity` is an empty list (`[]`).

Reply:
Cool! That simplifies things